src/daft-core/src/array/ops/any_value.rs

```diff
@@ -0,0 +1,104 @@
+use arrow2::{array::PrimitiveArray, bitmap::Bitmap};
+use common_error::DaftResult;
+
+use crate::{
+    array::{DataArray, FixedSizeListArray, ListArray, StructArray},
+    datatypes::*,
+};
+
+use super::{full::FullNull, DaftAnyValueAggable, GroupIndices};
+
+fn get_any_grouped_idx(
+    groups: &GroupIndices,
+    ignore_nulls: bool,
+    validity: Option<&Bitmap>,
+) -> UInt64Array {
+    let group_indices = if ignore_nulls && let Some(validity) = validity {
+        Box::new(PrimitiveArray::from_trusted_len_iter(groups.iter().map(|g| {
+            for i in g {
```

Comment:
you can rewrite this as `g.find(|i| validity.get_bit(*i as usize))`

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find